packages/components/src/components/post-avatar/post-avatar.tsx

```diff
@@ -0,0 +1,120 @@
+import { Component, h, Host, Prop, State, Watch } from '@stencil/core';
+import { version } from '@root/package.json';
+import { checkNonEmpty, checkOneOf } from '@/utils';
+import aiconizer from './aiconizer';
+
+// https://docs.gravatar.com/api/avatars/images/
+const GRAVATAR_DEFAULT = '404';
+const GRAVATAR_RATING = 'g';
+
+const GRAVATAR_BASE_URL = `https://www.gravatar.com/avatar/{email}?s={size}&d=${GRAVATAR_DEFAULT}&r=${GRAVATAR_RATING}`;
+
+enum AvatarType {
+  Gravatar = 'gravatar',
+  Initials = 'initials',
+  Null = null,
+}
+
+@Component({
+  tag: 'post-avatar',
+  styleUrl: 'post-avatar.scss',
+  shadow: true,
+})
+export class PostAvatar {
+  private static GRAVATAR_SIZES = {
+    large: 40,
+    small: 32,
+  };
+
+  /**
+   * Defines the size of the avatar.
+   */
+  @Prop() readonly size?: 'large' | 'small' = 'large';
+
+  /**
+   * Defines the users email address.
+   */
+  @Prop() readonly email?: string;
+
+  /**
+   * Defines the users firstname.
+   */
+  @Prop() readonly firstname!: string;
+
+  /**
+   * Defines the users lastname.
+   */
+  @Prop() readonly lastname?: string;
+
+  @State() avatarType: AvatarType = null;
+  @State() gravatarUrl = '';
+  @State() initials = '';
+
+  @Watch('firstname')
+  firstnameChanged() {
+    checkNonEmpty(this.firstname, 'The `firstname` property of the `post-avatar` is required!');
+  }
+
+  @Watch('size')
+  sizeChanged() {
+    checkOneOf(
+      this.size,
+      ['large', 'small'],
+      'The `size` property of the `post-avatar` must be either `large` or `small`.',
+    );
+  }
+
+  private async getAvatarType() {
+    let response: Response = new Response();
+
+    if (this.email) {
+      const email = await crypto.subtle
```

Comment:
Should we check for post addresses and try to display internal images when available? I think that would be a nice addition (and something the current KLP Login Widget is doing).

Reply:
Do you know, where the user pictues could come from then? I coulden't find anything in the klp-login-widget so far.

Reply:
True, I think the Intranet Header knows where to get the images.